pkg/loader/compose/compose.go

```diff
@@ -273,12 +322,126 @@ func loadPorts(composePorts []string) ([]kobject.Ports, error) {
 	return ports, nil
 }
 
-// LoadFile loads compose file into KomposeObject
+// LoadFile loads a compose file into KomposeObject
 func (c *Compose) LoadFile(files []string) (kobject.KomposeObject, error) {
+
+	// What will finally be returned
 	komposeObject := kobject.KomposeObject{
 		ServiceConfigs: make(map[string]kobject.ServiceConfig),
 		LoadedFrom:     "compose",
 	}
+
+	// Load the json / yaml file in order to get the version value
+	version, err := getVersionFromFile(files[0])
+	if err != nil {
+		return kobject.KomposeObject{}, errors.Wrap(err, "Unable to load yaml/json file for version parsing")
+	}
+	log.Debugf("Docker Compose version: %s", version)
+
+	// Convert based on version
+	switch version {
+	// Use libcompose for 1 or 2
+	// If blank, it's assumed it's 1 or 2
```

Comment:
No, in docker-compose and `docker stack deploy`, it always assumes v1 if version isn't included in the file